single-function/lru-cache/submission.js

```diff
@@ -0,0 +1,66 @@
+class DoubleLinkedList {
+  constructor(key, value) {
+    this.key = key;
+    this.value = value;
+  }
+}
+
+class LRUCache {
+  constructor(capacity) {
+    this.capacity = capacity;
+    this.cache = {};
+    let head = new DoubleLinkedList(0, 0);
+    let tail = new DoubleLinkedList(0, 0);
+    head.next = tail;
+    tail.prev = head;
+    this.head = head;
+    this.tail = tail;
+    this.count = 0;
+  }
+
+  deleteDoubleLinkedList(node) {
+    node.prev.next = node.next;
+    node.next.prev = node.prev;
+  }
+
+  addNodeToHead(node) {
+    node.next = this.head.next;
+    this.head.next.prev = node;
+    this.head.next = node;
+    node.prev = this.head;
+  }
+
+  get(key) {
+    if (this.cache[key]) {
+      return this.ejectLinkedList(this.cache[key]).value;
+    } else {
+      return -1;
```

Comment:
let's use names wherever we can, to keep things self-describing -- plz make this a named constant. What's this `-1` represent semantically.

Reply:
airBnB wants it to look like this:
```      const { value } = this.ejectLinkedList(this.cache[key]);
      return value;```

Reply:
-1 represents not found. I pulled that from reading about LRU on wiki. I should probably make that return false?